components/common/src/templating/hooks.rs

```diff
@@ -769,21 +827,21 @@ echo "The message is Hello"
             ServiceGroup::new(None, "dummy", "service", None).expect("couldn't create \
                                                                       ServiceGroup");
 
-        hook_output.stream_output::<InstallHook>(&service_group, &mut child);
+        hook_output.output_standard_streams::<InstallHook>(&service_group, &mut child);
 
-        let mut stdout = String::new();
-        hook_output.stdout()
-                   .unwrap()
-                   .read_to_string(&mut stdout)
-                   .expect("couldn't read stdout");
-        assert_eq!(stdout, "This is stdout\n");
+        let stdout = hook_output.stdout_str().expect("to get stdout string");
+        assert_eq!(stdout, "This is stdout\nThis is stdout line 2\n");
 
-        let mut stderr = String::new();
-        hook_output.stderr()
-                   .unwrap()
-                   .read_to_string(&mut stderr)
-                   .expect("couldn't read stderr");
-        assert_eq!(stderr, "This is stderr\n");
+        let stderr = hook_output.stderr_str().expect("to get stderr string");
+        assert_eq!(stderr,
+                   "This is stderr\nThis is stderr line 2\nThis is stderr line 3\n");
+
+        std::env::set_var(HookStandardStreamByteLimit::ENVVAR, "20");
```

Comment:
I love that we're testing this. To ensure that tests don't interfere with each other, it's best to use the `locked_env_var` macro when modifying environment variables in tests.